### PR TITLE
Fixes #1729: Fixed Appveyor CI setup for UNIX environments under Windows

### LINKED ISSUE #1729: Fix Appveyor CI for native Windows and Windows with UNIX-env
The manual-ci-run branch currently fails on Appveyor, plus the runtime environment used on Appveyor is incorrect.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

**Comment by andy-maier** [comment]
I found that there are some issues when using it on a local MSYS2 and CygWin. This definitely needs to be tested locally. One issue is a pip issue I have described in pypa/pip#6423.

File: docs/changes.rst [hunk 1]
```diff
@@ -39,6 +39,9 @@ Released: not yet
 
 * Test: Added and fixed profile definitions for end2end tests. (Issue #1714)
 
+* Test: Fixed Appveyor CI setup for UNIX environments under Windows
```
Comment (KSchopmeyer) [inline docs/changes.rst]: I think the the common acronym for this is WSL (Windows Subsystem for Linux). Probably more people will understand if we use that terminology.

Comment (andy-maier) [reply docs/changes.rst]: WSL is something different than what CygWin and MSYS2 provide: https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux. Neither CygWin nor MSYS2 register itself as a WSL, nor do they intend to serve as a runtime compatibility layer for running Linux executables in WLF format. Instead, they serve mostly as a development environment for executables to be run on native Windows (in addition to be run in their respective shells).

Comment (andy-maier) [reply docs/changes.rst]: Karl agreed in the slack channel. DONE.

**Comment by andy-maier** [comment]
I have updated the PR to remove MSYS2 from the set of environments to be tested.
I'm not quite sure how to go about it in the documentation.

**Comment by KSchopmeyer** [comment]
My first question is whether MSYS2 it a goal we need to achieve. 

NOTE: I made a mistake in first testing this and tested it against Microsoft WSL (https://docs.microsoft.com/en-us/windows/wsl/install-win10) which builds a linux subsystem into windows using one of the std linux flavors  (ex ubuntu,  within the windows file system but with a separate terminal for WSL and with the WSL filesystem hitten within the \user\<name>. I am not sure if this is worth adding to our tests or not at this point.

**Comment by andy-maier** [comment]
On testing WSL: I don't have Windows 10 at home, nor does Appveyor.

**Comment by andy-maier** [comment]
In a call last week, we agreed to remove the support for MSYS2 again.

**Comment by andy-maier** [comment]
The latest update removes MSYS2 support.
The PR is now ready for final review.

File: docs/changes.rst [hunk 2]
```diff
@@ -50,6 +50,9 @@ Released: not yet
   property name in the returned instance. Works in the conn_lite=True mode but
   fails in conn_lite=False mode because the test was case insensitive.
 
+* Test: Fixed Appveyor CI setup for UNIX environments under Windows
```
Comment (KSchopmeyer) [inline docs/changes.rst]: Shouldn't this be unix-like to be consistent with other wording

File: docs/intro.rst [hunk 3]
```diff
@@ -305,7 +305,7 @@ present in the PATH.
   * ``tar``
 
   These commands can be made available in the PATH via a UNIX-like environment
-  such as `CygWin`_, `MSYS2`_, Babun, or Gow. If they are not all available in
+  such as `CygWin`_, MSYS2, Babun, or Gow. If they are not all available in
```
Comment (KSchopmeyer) [inline docs/intro.rst]: Chocolatey is mentioned just above here so shouldn't the note about this being a preq be mentioned there and maybe it should be in the list above this at about 305

File: docs/changes.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-maier) [reply docs/changes.rst]: DONE

File: docs/intro.rst (hunk 3 above)
Comment (andy-maier) [reply docs/intro.rst]: DONE, but differently:

There are two difficulties with your proposal:
* The `choco` comamnd is only needed if these other commands are not in the PATH
* While these other commands can be made available by the user via a UNIX-like environment, that is not the case for the `choco` command.

Therefore, I don't think we should simply add the `choco` command to the list above.

Instead, I have tried addressing this by changing the paragraph to:

> These commands can either be made available via a UNIX-like environment such as `CygWin`_, MSYS2, Babun, or Gow, or if they are not available in the PATH, they will be installed using the `Chocolatey package manager`_ by invoking its ``choco`` command.

**Comment by andy-maier** [comment]
Addressed both comments, and assuming your agreement to the one that I solved differently.

**Pull request merged**